### ISSUE TITLE
Request to add Graph Normalizing Flows

### ISSUE THREAD

**Issue by johncava** [rusty1s/pytorch_geometric#897]
## 🚀 Feature

**Original Paper**
https://arxiv.org/abs/1905.13177

**Original Paper TensorFlow Repo**
https://github.com/jliu/graph-normalizing-flows

**Motivation**
A very interesting paper proposing an alternative graph generative model to GraphRNN using normalizing flows. I think it is beneficial to pytorch geometric as other papers such as https://arxiv.org/pdf/1909.13521.pdf have begun to add other flavors of normalizing flows for graph generation.

Thanks!

**Comment by rusty1s** [comment]
I see what I can do :)

**Comment by murnanedaniel** [comment]
@rusty1s Have you had a look at this yet? If not, I was interested in getting some normalizing flow module off the ground. But I don't want to re-invent the wheel if you've already made progress.

**Comment by rusty1s** [comment]
That would be great, I would very much appreciate this! Please feel free to send a PR :)

**Comment by prateeky2806** [comment]
Hi @murnanedaniel, I plan to work on graph normalizing flows in my research project. I just wanted to check if the this is coming soon into pytorch-geometric or not.

**Comment by murnanedaniel** [comment]
Hi @prateeky2806, I didn't end up implementing this. My goal was to use the _reversible_ nature of normalising flows in order to save memory when training, but the gradient checkpointing features in Pytorch ended up achieving this more easily. So if you would like to implement this, you should go right ahead!

**Comment by thegodone** [comment]
Hi guys any update in this PR ?
